tests/functional/spec/timeout_modal.spec.js

```diff
@@ -0,0 +1,11 @@
+describe("Timeout Modal", () => {
+  beforeEach("Open Survey", () => {
+    browser.openQuestionnaire("test_timeout_modal.json");
+  });
+
+  it("Given I am completing the survey, when session time out is set to same amount as the showModalTimeInSeconds of timeout modal I will be able to see timeout modal with the option to extend the session instantly", () => {
```

Comment:
The focus of the test here should be that, there is only 60 seconds until session expiry. We may want to set it to like 75 seconds and ensure there is no popup until we hit the 60-second mark. Since it's client side, you will have to be estimating the time left as it won't be an exact science.

Reply:
To properly test, the timeout needs to be like at least > 120 seconds, as we only store the updated session if it has been at least 60 seconds. Otherwise, if you were to use say 75 second, then after 15 seconds you would get the pop-up modal, but you can't extend the session as the session is only 15 seconds old. So clicking `Continue` on the modal would only work when the countdown is at like 14 seconds left, at which point our session is now >60 seconds old.

Reply:
I've extended the timeout to exactly 120 seconds, which works in new functional tests, all the scenarios we discussed are covered (`Continue` button, post submission, page back in focus).